mreg_cli/commands/host_submodules/core.py

```diff
@@ -140,34 +141,81 @@ def add(args: argparse.Namespace) -> None:
             metavar="NAME/IP",
         ),
         Flag("-force", action="store_true", description="Enable force."),
+        Flag(
+            "-override",
+            short_desc="Comma separated override list, requires -force.",
+            description=(
+                "Comma separated overrides for forced removal. Requires -force."
+                "Accepted overrides: 'cname', 'ipadress', 'mx', 'srv', 'ptr', 'naptr'."
+                "Example usage: '-override cname,ipaddress,mx'"
+            ),
+            metavar="OVERRIDE",
+        ),
     ],
 )
 def remove(args: argparse.Namespace) -> None:
     """Remove host.
 
-    :param args: argparse.Namespace (name, force)
+    :param args: argparse.Namespace (name, force, override)
     """
     # Get host info or raise exception
     info = host_info_by_name_or_ip(args.name)
+    overrides: List[str] = args.override.split(",") if args.override else []
+
+    accepted_overrides = ["cname", "ipaddress", "mx", "srv", "ptr", "naptr"]
+    for override in overrides:
+        if override not in accepted_overrides:
+            cli_warning(f"Invalid override: {override}. Accepted overrides: {accepted_overrides}")
 
-    warn_msg = ""
+    def forced(override_required: Optional[str] = None) -> bool:
+        # If we require an override, check if it's in the list of provided overrides.
+        if override_required:
+            return override_required in overrides
+
+        # We didn't require an override, so we only need to check for force.
+        if args.force:
+            return True
+
+        # And the fallback is "no".
+        return False
+
+    warnings: List[str] = []
     # Require force if host has any cnames.
-    cnames = info["cnames"]
-    if len(cnames):
-        if not args.force:
-            warn_msg += "{} cnames. ".format(len(cnames))
+    if info["cnames"] and not args.force:
+        warnings.append(f"  {len(info['cnames'])} cnames, override with 'cname'")
+        for cname in info["cnames"]:
+            warnings.append(f"    - {cname['name']}")
+
+    # Require force if host has multiple A/AAAA records and they are not in the same VLAN.
+    if len(info["ipaddresses"]) > 1:
+        same_vlan = ips_are_in_same_vlan([ip["ipaddress"] for ip in info["ipaddresses"]])
+
+        if same_vlan and not forced():
+            warnings.append("  multiple ipaddresses on the same VLAN. Must use 'force'.")
```

Comment:
Should this be more clear that we are referring to the option `-force`, not some argument?

```suggestion
            warnings.append("  multiple ipaddresses on the same VLAN. Must use -force.")
```